pkg/rulefmt/rulefmt.go

```diff
@@ -101,55 +116,104 @@ type Rule struct {
 	Annotations map[string]string `yaml:"annotations,omitempty"`
 }
 
+// RuleNode adds yaml.v3 layer to support line and column outputs for invalid rules.
+type RuleNode struct {
+	Record      yaml.Node         `yaml:"record,omitempty"`
+	Alert       yaml.Node         `yaml:"alert,omitempty"`
+	Expr        yaml.Node         `yaml:"expr"`
+	For         model.Duration    `yaml:"for,omitempty"`
+	Labels      map[string]string `yaml:"labels,omitempty"`
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
 // Validate the rule and return a list of encountered errors.
-func (r *Rule) Validate() (errs []error) {
-	if r.Record != "" && r.Alert != "" {
-		errs = append(errs, errors.Errorf("only one of 'record' and 'alert' must be set"))
+func (r *RuleNode) Validate() (errs []WrapError) {
```

Comment:
You only use this as an `error`

Reply:
I am wrapping this error with the node values so that we can implement according to what slrbfts said. Do you mean to remove the wrapper logic and use the earlier error logic?

Reply:
No, I mean the return type here should be a slice of error.

Reply:
@brian-brazil  If I return a slice of error here, then the suggestion given by slrbtfs would not be possible. Please correct me if I am wrong.

Reply:
@brian-brazil any suggestions?

Reply:
Error is an interface, it's weird to refer to things as errors that don't meet that interface. So I suggest adjusting things so that your errors are errors.

Reply:
> @brian-brazil If I return a slice of error here, then the suggestion given by slrbtfs would not be possible. Please correct me if I am wrong.

It is possible.

You can return an error and then convert it back to the underlying concrete type when needed.

https://tour.golang.org/methods/15

Reply:
> Error is an interface, it's weird to refer to things as errors that don't meet that interface. So I suggest adjusting things so that your errors are errors.

With the current design, the errors are being returned after unwrapping, done by the Error interface function. Only after that, it is an error. I think the `errs` is leading to confusion even though its a slice of wrapped error along with its node value. Is this any case?